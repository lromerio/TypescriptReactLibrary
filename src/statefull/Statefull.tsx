import * as React from "react";
import {style} from "typestyle/lib";
import {Base} from "../styles";
import "../styles.css";

export interface IStatefullProps {
    id?: string;
    content: string;
}

interface IStatefullDefaultProps {
    id?: string;
}

interface IStatefullState {
    myState: string;
}

const  Statefull: React.ComponentClass<IStatefullProps> =
    class extends React.Component<IStatefullProps & IStatefullDefaultProps, IStatefullState> {

        public static defaultProps: IStatefullDefaultProps = {
            id: "my_id",
        };

        constructor(props: IStatefullProps & IStatefullDefaultProps) {
            super(props);

            this.state = {myState: "useless"};
        }

        public render() {
            return <span id={this.props.id} className={styles}>{this.props.content}</span>;
        }
    };

/**
 * Styles (CSS like) of the component.
 */
const styles = style(Base.text, {
    color: "blue",
    fontSize: "50px",
    padding: "20px 20px 20px 0px",
});

export default Statefull;
