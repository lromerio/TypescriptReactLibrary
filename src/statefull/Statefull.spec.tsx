import "raf/polyfill";
import "jsdom-global/register"

import * as React from "react"
import {configure, mount, shallow} from "enzyme"
import * as Adapter from "enzyme-adapter-react-16"
import toJson from "enzyme-to-json"

import Statefull from "."


describe("<Statefull />", () => {

    const text = "Test"

    const comp = <Statefull content={text} />

    configure({ adapter: new Adapter() })

    it("should render", () => {
        const wrapper = mount(comp)
        expect(toJson(wrapper)).toMatchSnapshot()
        wrapper.unmount()
    })

    it("should update with props", () => {
        const newText = "New Text"
        
        const wrapper = shallow(comp)

        expect(wrapper.find("span").text()).toBe(text)
        wrapper.setProps({content: newText })
        expect(wrapper.find("span").text()).toBe(newText)

        wrapper.unmount()
    })
})