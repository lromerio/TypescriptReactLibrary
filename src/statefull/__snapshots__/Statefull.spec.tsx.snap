// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Statefull /> should render 1`] = `
<class_1
  content="Test"
  id="my_id"
>
  <span
    className="fjy8pip"
    id="my_id"
  >
    Test
  </span>
</class_1>
`;
